pkg/repo/index.go

```diff
@@ -99,6 +99,17 @@ func (index *Index) AddEntry(chartVersion *helm_repo.ChartVersion) {
 	if _, ok := index.Entries[chartVersion.Name]; !ok {
 		index.Entries[chartVersion.Name] = helm_repo.ChartVersions{}
 	}
+	//
+	entries := index.Entries[chartVersion.Name]
+	l := len(entries)
+	for i := 1; i <= 5 && l-i >= 0; i++ {
+		cv := entries[l-i]
+		if cv.Version == chartVersion.Version {
+			index.setChartURL(chartVersion)
+			entries[l-i] = chartVersion
+			return
+		}
+	}
```

Comment:
Could you help me understand what this is doing? What is the significance of the number 5?

Reply:
If someone push the same chart to `Chartmuseum` and `AllowOverwrite` is true, cache will  exist duplicate version info, like this
```yaml
  - created: "2019-10-21T09:46:09.216Z"
    digest: 963a506b5523966b993fe68c118d438402121b0d53cb412b9fc50328a428bc6e
    name: account1
    urls:
    - charts/account1-0.0.1-595d1a1bdb2d00edf976c46cde9c7562fab490ba.tgz
    version: 0.0.1-595d1a1bdb2d00edf976c46cde9c7562fab490ba
  - created: "2019-10-21T09:46:09.216Z"
    digest: 963a506b5523966b993fe68c118d438402121b0d53cb412b9fc50328a428bc6e
    name: account1
    urls:
    - charts/account1-0.0.1-595d1a1bdb2d00edf976c46cde9c7562fab490ba.tgz
    version: 0.0.1-595d1a1bdb2d00edf976c46cde9c7562fab490ba
```
It's a lazy solution to look for 5 versions from back to front. If none of the 5 versions are same to the new pushing version, I think it is a new version and not has been pushed. If there is a same version, I would update it. Because every new version will just be `append` to the end, I thought this solution is safe. Actually just comparing with the last one version is enough. For insurance purposes, I look for `5` versions from back to front and compare them with the new pushing version.

 ```go
index.Entries[chartVersion.Name] = append(index.Entries[chartVersion.Name], chartVersion)
```
I really hope you have a better way.